src/environment/providers/util.ts

```diff
@@ -8,3 +8,8 @@ export function flattenAndCompact<T>(value: (T | T[] | null)[] | null): T[] | nu
     const merged = flatten(compact(value))
     return merged.length === 0 ? null : merged
 }
+
+/** Flattens and compacts the argument. If it is null or if the result is empty, it returns null. */
+export function flattenAndCompactNonNull<T>(value: (T | T[] | null)[] | null): T[] {
```

Comment:
This is only used in 1 place. Don't extract it to another module. You probably shouldn't even extract it to a function, you should just inline it.